hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -608,6 +608,14 @@ public boolean populateMetaFields() {
         HoodieTableConfig.HOODIE_POPULATE_META_FIELDS.defaultValue()));
   }
 
+  public String getSimpleRecordKeyField() {
+    return getString(HoodieTableConfig.HOODIE_TABLE_SIMPLE_RECORDKEY_FIELD);
```

Comment:
I do see we already have two configs in table props to hold record key and partition path from sql layer. But I am not very sure if it works across all layers. i.e. data source, write client etc. in other words, not sure if I can re-use the same variables to store record keys and partition paths. 
Open to discuss and decide on how to go about this.